crates/electrum/src/bdk_electrum_client.rs

```diff
@@ -579,8 +579,18 @@ impl<E: ElectrumApi> BdkElectrumClient<E> {
                     let outpoint = vin.previous_output;
                     let vout = outpoint.vout;
                     let prev_tx = self.fetch_tx(outpoint.txid)?;
-                    let txout = prev_tx.output[vout as usize].clone();
-                    let _ = tx_update.txouts.insert(outpoint, txout);
+                    // Ensure server returns the expected txout.
+                    let txout = prev_tx
+                        .output
+                        .get(vout as usize)
+                        .ok_or_else(|| {
+                            electrum_client::Error::Message(format!(
+                                "server returned tx {} with no vout",
+                                prev_tx.compute_txid()
```

Comment:
```suggestion
                                "prevout {outpoint} does not exist"
```

Not sure what "no vout" refers to here.